src/background_workers.c

```diff
@@ -238,11 +257,10 @@ static RedisAI_RunInfo **_BGThread_BatchOperations(RunQueueInfo *run_queue_info,
         // there's anything else to batch
         current_batchsize += next_batchsize;
 
-        // If minbatchsize hasn't been set, or if the current batch
-        // size exceeds the minimum batch size already, then we're done.
+        // If batch size exceeds the minimum batch size already, then we're done.
         // Otherwise, if minbatchsize was set and the size wasn't reached,
         // loop until there's something new on the queue
-        if (minbatchsize == 0 || current_batchsize >= minbatchsize) {
+        if (minbatchsize != 0 && current_batchsize >= minbatchsize) {
```

Comment:
Do we really need to check it? why stopping at minbatchsize if we can extend the batch even further (untill it reaches batchsize)?

Reply:
good point